File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Shearer.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;
import org.bukkit.entity.Sheep;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Shearer implements Listener {
    @EventHandler
    public void onInteract(final PlayerInteractEvent playerInteractEvent) {
        final Player player = playerInteractEvent.getPlayer();
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "shearer"));
        if (playerInteractEvent.getAction() == Action.LEFT_CLICK_AIR && player.getItemInHand() != null && player.getItemInHand().hasItemMeta() && player.getItemInHand().getItemMeta().hasLore() && player.getItemInHand().getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
            final int int1 = SettingsManager.enchant.getInt("shearer.level_I.radius");
            for (final Entity entity : player.getNearbyEntities((double) int1, (double) int1, (double) int1)) {
                if (entity instanceof Sheep) {
                    final Sheep sheep = (Sheep) entity;
                    if (sheep.isSheared()) {
                        continue;
                    }
                    sheep.setSheared(true);
                    sheep.getWorld().dropItem(sheep.getLocation(), new ItemStack(Material.WOOL, 1, (short) sheep.getColor().getWoolData()));
                }
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Shearer.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.Material
import org.bukkit.entity.Sheep
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.inventory.ItemStack
import org.pixeltime.enchantmentsenhance.listener.EnchantmentListener
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Shearer : EnchantmentListener() {
    override fun lang(): Array<String> {
        return arrayOf("剪裁")
    }

    @EventHandler(ignoreCancelled = true, priority = EventPriority.MONITOR)
    fun onInteract(playerInteractEvent: PlayerInteractEvent) {
        val player = playerInteractEvent.player

        if (playerInteractEvent.action == Action.LEFT_CLICK_AIR) {
            val level = IM.getHighestLevel(player, this.name())
            if (level > 0) {
                val int1 = SettingsManager.enchant.getInt("shearer.$level.radius")
                for (entity in player.getNearbyEntities(int1.toDouble(), int1.toDouble(), int1.toDouble())) {
                    if (entity is Sheep) {
                        if (!entity.isSheared) {
                            entity.isSheared = true
                            entity.world.dropItem(entity.location, ItemStack(Material.WOOL, 1, entity.color.woolData.toShort()))
                        }
                    }
                }
            }
        }
    }
}
